docs/markdown/templating/00-TITLE.md

```diff
@@ -0,0 +1,115 @@
+<!-- .slide: class="transition" -->
+
+# Templating
+
+##==##
+
+# Templates
+
+<br/>
+
+* Les templates favorisent la mise en forme du code en utilisant des fichiers de variables .tfvars qui seront chargés par Terraform au runtime (plan ou apply).
+* Le fichier .tfvars peut contenir:
+  * des variables simples
+  * des listes
+  * des maps
+  * des listes de map ou des listes de listes
+* La variable doit être définie "vide" dans le fichier de variable.
+* Il y a plusieurs types de templating utilisable avec Terraform : 
+  * Terragrunt
+  * Handlebars
+  * terraform propose également son propre modèle de templating (c'est sur celui-ci que nous nous concentrerons)
+
+
+##==##
+<!-- .slide: -->
+
+# Templates
+
+## Modules
+
+Dans la partie précédente, nous avons vu comment appeler un module stocké sur un repo.
+Cependant, il est possible que le module que vous souhaitez utiliser ne réponde pas entièrement à vos besoin. Dans ces cas là, il devient necéssaire de créer votre propre module.
+Une fois le module créé, vous pouvez le conserver en local ou bien le publier sur votre propre repo.
+
+##==##
+<!-- .slide: class="with-code-bg-dark" -->
+
+# Templates
+
+=> *cat cloud_sql/main.tf*
+```hcl
+resource "google_sql_database" "database" {
+  count     = length(var.database)
```

Comment:
Hello,
Il serait plus pertinent de passer sur un model for_each pour éviter de ré-indexer lors de la suppression d'un élément (plus en phase avec le questionnaire)

Reply:
Ca revient au même, ou presque.
Avec length(var.xxx), tu peux itérer sur absolument tout le contenu de la variable (dans l'esprit des listes de map).
Par contre, avec for_each, tu sélectionne les variables sur lesquelles tu vas itérer.
Les deux approches se valent.
J'ai plus tendances à utiliser for_each sur des blocs dynamiques (même quand je ne fais pas de magie noire avec terraform)